packages/core/http/core-http-router-server-internal/src/router.test.ts

```diff
@@ -22,6 +22,53 @@ const routerOptions: RouterOptions = {
 };
 
 describe('Router', () => {
+  describe('#getRoutes', () => {
+    it('returns expected route metadata', () => {
+      const router = new Router('', logger, enhanceWithContext, routerOptions);
+      const validation = schema.object({ foo: schema.string() });
+      router.post(
+        { path: '/', validate: { body: validation, query: validation, params: validation } },
+        (context, req, res) => res.ok()
+      );
+      const routes = router.getRoutes();
+      expect(routes).toHaveLength(1);
+      const [route] = routes;
+      expect(route).toMatchObject({
+        handler: expect.any(Function),
+        method: 'post',
+        path: '/',
+        validationSchemas: { body: validation, query: validation, params: validation },
+        isVersioned: false,
```

Comment:
Mainly making sure that the above two fields are included as expected, will be used by OAS generation script